tutorial/openmp.md

```diff
@@ -0,0 +1,583 @@
+
+n = 1337
+arr = zeros((n))
+arr_2 = zeros((n))
+#$ omp parallel num_threads(4) private(i, j) shared(arr)
+
+#$ omp for
+for i in range(0, n):
+  arr[i] = i
+#$ omp barrier
+#$ omp for
+for j in range(0, n):
+  arr_2[j] = arr[j] * 2
+
+#$ omp end parallel
+print(sum(arr_2))
+```
+
+The output of this program is:
+```shell
+❯ pyccel omp_test.py --openmp
+❯ ./omp_test
+1786232
+```
+
+### Masked Construct
+
+#### Syntax
+
+```python
+#$ omp masked [ filter(integer-expression) ]
+structured-block
+#$ omp end masked
+```
+
+#### Example
+
+The ``` #$ omp masked ``` pragma is used here to specify a structured block that is executed by a subset of the threads of the current team.
+```python
+result = 0
+#$ omp parallel num_threads(4)
+#$ omp masked
+result = result + 1
+#$ omp end masked
+#$ omp end parallel
+print("result :", result)
+```
+
+The output of this program is:
+```shell
+❯ pyccel omp_test.py --openmp
+❯ ./omp_test
+result : 1
+```
+
+### Taskloop/Atomic Construct
+
+#### Syntax Taskloop
+
+```python
+#$ omp taskloop [clause[ [,]clause] ... ]
+for-loops
+```
+
+#### Syntax Atomic
+
+```python
+#$ omp atomic [clause[ [,]clause] ... ]
+structured-block
+#$ omp end atomic
+```
+
+#### Example
+The ``` #$ omp taskloop ``` construct specifies that the iterations of one or more associated loops will be executed in parallel using explicit tasks.\
+The ``` #$ omp atomic ``` is used to ensure that a specific storage location is accessed atomically; which prevent the possibility of multiple, simultaneous reading and writing of threads.
+```python
+from pyccel.stdlib.internal.openmp import omp_get_thread_num
+
+x1 = 0
+x2 = 0
+#$ omp parallel shared(x1,x2) num_threads(2)
+
+#$ omp taskloop
+for i in range(0, 100):
+  #$ omp atomic
+  x1 = x1 + 1 #Will be executed (100 x 2) times.
+  #$ omp end atomic
+
+#$ omp single
+#$ omp taskloop
+for i in range(0, 100):
+  #$ omp atomic
+  x2 = x2 + 1 #Will be executed (100) times.
+  #$ omp end atomic
+#$ omp end single
+
+#$ omp end parallel
+print("x1 : ", x1);
+print("x2 : ", x2);
+```
+
+The output of this program is (you may get a different output, but the sum must be the same for each thread):
+```shell
+❯ pyccel omp_test.py --openmp
+❯ ./omp_test
+x1 : 200
+x2 : 100
+```
+
+### SIMD Construct
+
+#### Syntax
+
+```python
+#$ omp simd [clause[ [,]clause] ... ]
+loop-nest
+```
+
+#### Example
+
+The ``` #$ omp simd ``` pragma is used to transform the loop into a loop that will be executed concurrently using Single Instruction Multiple Data (SIMD) instructions.
+```python
+from numpy import zeros
+result = 0
+n = 1337
+arr = zeros(n, dtype=int)
+#$ omp parallel num_threads(4)
+#$ omp simd
+for i in range(0, n):
+  arr[i] = i
+#$ omp end parallel
+for i in range(0, n):
+  result = result + arr[i]
+print("Result:", result)
+```
+
+The output of this program is:
+```shell
+❯ pyccel omp_test.py --openmp
+❯ ./omp_test
+Result: 893116
+```
+
+### Task / Taskwait Construct
+
+#### Syntax Task Construct
```

Comment:
As said above, this should read

`#### Syntax of *task*`

which is displayed as

#### Syntax of *task*